### PR TITLE
Security docs

### PR DESCRIPTION
**Pull request by ru-fu**
Clean up and extend the contents of SECURITY.md and doc/security.md.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -41,22 +41,20 @@ More instructions on installing LXD for a wide variety of Linux distributions an
 
 To install LXD from source, see [Installing LXD](doc/installing.md) in the documentation.
 
-<!-- Include start security -->
-
 ## Security
-LXD, similar to other container and VM managers provides a UNIX socket for local communication.
 
-**WARNING**: Anyone with access to that socket can fully control LXD, which includes
```
Comment (stgraber) [inline README.md:49]: I think I'd like to keep the prominent warning in the README.md, this is more visible than the bulletpoints and I think needs to be seen without having to go to the separate security doc.

File: doc/authentication.md [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+- auditor: Read-only access to the project
+- user: Ability to do normal life cycle actions (start, stop, ...),
+        execute commands in the instances, attach to console, manage snapshots, ...
+- operator: All of the above + the ability to create, re-configure and
+            delete instances and images
+- admin: All of the above + the ability to reconfigure the project itself
+
+```{important}
+Currently, only the `auditor` and the `user` roles are suitable for users that you wouldn't trust with root access to the host.
+```
```
Comment (stgraber) [inline doc/authentication.md:140]: I a follow-up change, we should probably rephrase that as it's no longer quite true.
The statement remains valid unless the project is restricted, restricted projects can be made to be safe for the `operator` role too. `admin` remains dangerous as it allows reconfiguring the project and therefore turning off restrictions.

File: doc/authentication.md [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+
+A remote add operation should therefore be implemented like this:
+
+1. Call GET /1.0.
+1. Ask the user to confirm the fingerprint.
+1. Look at the dict received from the server. If "auth" is "untrusted":
+
+   1. Ask the user for the server's password.
+   1. Do a `POST` to `/1.0/certificates`.
+   1. Call `/1.0` again to check that the client is now trusted.
```
Comment (stgraber) [inline doc/authentication.md:79]: In a future update of this doc, we probably ought to get rid of this. We haven't really been providing such API workflow/examples throughout the doc so having this one here feels weird. If we find the need, we could always add something like this back as a howto.

File: doc/authentication.md [hunk 4]
```diff
@@ -0,0 +1,162 @@
...
+#### Adding client certificates using a trust password
+
+To allow establishing a new trust relationship from the client side, you must set a trust password (`core.trust_password`, see {doc}`server`) for the server. Clients can then add their own certificate to the server's trust store by providing the trust password.
+
+The workflow to authenticate with the server is similar to that of SSH, where an initial connection to an unknown server triggers a prompt:
+
+1. When the user adds a server with `lxc remote add`, the server is contacted over HTTPS, its certificate is downloaded and the fingerprint is shown to the user.
+1. The user is asked to confirm that this is indeed the server's fingerprint, which they can manually check by connecting to the server or by asking someone with access to the server to run the info command and compare the fingerprints.
+1. The user enters the trust password for the server.
+1. If the trust password matches, the client certificate is added to the server's trust store. The client can now connect to the server without having to provide any additional credentials.
```
Comment (stgraber) [inline doc/authentication.md:69]: The first two steps here also apply to the `trusted certificates` case above, not sure if we should duplicate the content or re-shuffle things a bit so this bit goes before the two different authentication methods.

Just to keep in mind for a later edit, not a blocker for this PR.

File: doc/security.md [hunk 5]
```diff
@@ -1,289 +1,129 @@
...
-the network on a TLS socket.
+% Include content from [../README.md](../README.md)
+```{include} ../README.md
+    :start-after: <!-- Include start security -->
+    :end-before: <!-- Include end security -->
+```
+
+See the following sections for detailed information.
 
-**WARNING**: Local access to LXD through the UNIX socket always grants
```
Comment (stgraber) [inline doc/security.md:9]: So we'll definitely want some version of this warning to be prominently visible on this page. Re-introducing it in the main README.md as already mentioned will hopefully do the trick through the include.

File: doc/authentication.md (hunk 2 above)
Comment (ru-fu) [reply doc/authentication.md:140]: Updated. Can we clarify what we mean with "restricted project"? Is this the same kind of restriction as mentioned in line 46 (https://github.com/lxc/lxd/pull/9787/files#diff-a781e12280f919cbdeeb19978c38c25508af7dbc1decd3f00d94863ec27bcdffR46)?

File: README.md (hunk 1 above)
Comment (ru-fu) [reply README.md:49]: Added back.

**Comment by ru-fu** [comment]
Thanks for the review, @stgraber !
I added a commit with updates, that should be easier to review - I can squash it before merging.

File: doc/authentication.md (hunk 2 above)
Comment (stgraber) [reply doc/authentication.md]: Yeah, it refers to https://linuxcontainers.org/lxd/docs/master/projects/#project-restrictions

Such projects don't have access (unless specifically configured by the administrator) to any of those LXD features that could allow taking over the whole system.

**Comment by stgraber** [comment]
@ru-fu looks good to me!

Would indeed be great if you could rebase the branch to apply those changes directly to the applicable commits, then we can merge it.

Thanks!

File: doc/authentication.md (hunk 2 above)
Comment (ru-fu) [reply doc/authentication.md]: Great, I added a link.

**Pull request merged**